### DISCUSSION TITLE
(Build) Dependency on mkdocstrings <= 0.17.0

### DISCUSSION THREAD

**Discussion by tijuca** [netbox-community/netbox#9141] [General]
Hi,

since some time I'm working on packaging all the outstanding Python libraries for Debian so NetBox can be build within Debian as an binary package and I'm happy with the current state of this.
By version 3.2 of NetBox there are again some new packages required to get the documentation build. One of them is `mkdocstrings` which is pinned to a version <= 0.17.0.

So hopefully a simple question, why is this version requirement set to this version? Why not set to something "greater or equal"? I'm asking because `mkdocstrings` 0.17.0 has some other indirect dependencies than the current version 0.18.1 which I need to package first before I can introduce `mkdocstrings` as an package into Debian.

**Comment by jeremystretch** [comment]
> why is this version requirement set to this version?

NetBox pins this version because `mkdocstrings` introduced an [experimental Python handler](https://mkdocstrings.github.io/python/) in v0.18, which currently has some problems rendering our docs. While it's possible to install a newer version and use the legacy handler, it's a bit confusing. Since there's nothing in the newer releases (yet) that we really need, I've stuck with v0.17 until I can dedicate some time to figuring out the issues.

**Comment by tijuca** [comment]
Thanks!

It's no problem for me to package 0.17.0 instead of 0.18.0. But it would be a problem if I first package the higher version and then need to go back to 0.17.0.
And 0.17.0 has even less indirect package deps which decreases the amount of required packaging work now.

**Comment by tijuca** [comment]
O.k., just a heads up.
I'm now able to build the NetBox MkDocs based documentation by only using Debian packages of the required tools and related libraries. Preparing a final NetBox Debian package will still take some more time.
